test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/StatusRegistryDataStoreTests.cs

```diff
@@ -0,0 +1,111 @@
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Tests.Integration.Persistence
+{
+    [FhirStorageTestsFixtureArgumentSets(DataStore.All)]
+    public class StatusRegistryDataStoreTests : IClassFixture<FhirStorageTestsFixture>
+    {
+        private readonly FhirStorageTestsFixture _fixture;
+        private readonly IFhirStorageTestHelper _testHelper;
+
+        public StatusRegistryDataStoreTests(FhirStorageTestsFixture fixture)
+        {
+            _fixture = fixture;
+            _testHelper = fixture.TestHelper;
+        }
+
+        [Fact]
+        public async Task GivenAStatusRegistry_WhenGettingStatuses_ThenTheStatusesAreRetrieved()
+        {
+            IReadOnlyCollection<ResourceSearchParameterStatus> expectedStatuses = await _fixture.FilebasedStatusRegistryDataStore.GetSearchParameterStatuses();
+            IReadOnlyCollection<ResourceSearchParameterStatus> actualStatuses = await _fixture.StatusRegistryDataStore.GetSearchParameterStatuses();
+
+            ValidateSearchParameterStatuses(expectedStatuses, actualStatuses);
+        }
+
+        [Fact]
+        public async Task GivenAStatusRegistry_WhenUpsertingNewStatuses_ThenTheStatusesAreAdded()
+        {
+            string statusName1 = "http://hl7.org/fhir/SearchParameter/Test-1";
+            string statusName2 = "http://hl7.org/fhir/SearchParameter/Test-2";
+
+            var status1 = new ResourceSearchParameterStatus
+            {
+                Uri = new Uri(statusName1), Status = SearchParameterStatus.Disabled, IsPartiallySupported = false,
+            };
+
+            var status2 = new ResourceSearchParameterStatus
+            {
+                Uri = new Uri(statusName2), Status = SearchParameterStatus.Disabled, IsPartiallySupported = false,
+            };
+
+            IReadOnlyCollection<ResourceSearchParameterStatus> readonlyStatusesBeforeUpsert = await _fixture.StatusRegistryDataStore.GetSearchParameterStatuses();
+            var expectedStatuses = readonlyStatusesBeforeUpsert.ToList();
+            expectedStatuses.Add(status1);
+            expectedStatuses.Add(status2);
+
+            var statusesToUpsert = new List<ResourceSearchParameterStatus> { status1, status2 };
+
+            try
+            {
+                await _fixture.StatusRegistryDataStore.UpsertStatuses(statusesToUpsert);
+
+                IReadOnlyCollection<ResourceSearchParameterStatus> actualStatuses = await _fixture.StatusRegistryDataStore.GetSearchParameterStatuses();
+
+                ValidateSearchParameterStatuses(expectedStatuses, actualStatuses);
+            }
+            finally
+            {
+                await _testHelper.DeleteSearchParameterStatusAsync(statusName1);
+                await _testHelper.DeleteSearchParameterStatusAsync(statusName2);
+            }
+        }
+
+        [Fact]
+        public async Task GivenAStatusRegistry_WhenUpsertingExistingStatuses_ThenTheStatusesAreNotAdded()
```

Comment:
can we verify and rename this to ThenTheStatusesAreNotAdded -> ThenTheExistingStatusesAreUpdated

Reply:
Good idea, thanks. Fixed in 94ea3cb.